lib/src/manup_delegate.dart

```diff
@@ -0,0 +1,20 @@
+part of manup;
+
+/// `ManupDelegate` class has required methods.
+/// Need  `appContext` and `shouldShowAlert` to show dialog.
+/// Other methods are optional.
+/// Default implemetation is in `ManupDelegateMixin` file.
+abstract class ManupDelegate {
+  // required
+  BuildContext get appContext;
+  http.Client get httpClient;
```

Comment:
Is this required externally?

Reply:
Yes as we close the connection once we fetch the data and could not reopen connection. so we need new client every time,

Reply:
I am confused still - Do we need the HTTP client to live externally from the ManUp service, or could one live inside the `getMetadata` function? If we need to recreate all the time, why not create there - though I would think that the HTTP client would live in the service and be created in the constructor